src/Engine/Utils.java

```diff
@@ -571,10 +570,13 @@ public static void sortLocationsByTravelTime(Unit unit, ArrayList<XYCoord> mapLo
   public static boolean pathCollides(GameMap map, Unit unit, Path path)
   {
     boolean result = false;
-    for( PathNode point : path.getWaypoints() )
+    boolean includeOccupiedSpaces = true; // Shouldn't matter, as we don't invoke canEnd()
+    FloodFillFunctor fff = unit.getMoveFunctor(includeOccupiedSpaces);
+    for( int i = 1; i < path.getPathLength(); ++i)
     {
-      Unit obstacle = map.getLocation(point.x, point.y).getResident();
-      if( null != obstacle && unit.CO.isEnemy(obstacle.CO) )
+      XYCoord from = path.getWaypoint(i-1).GetCoordinates();
+      XYCoord to   = path.getWaypoint( i ).GetCoordinates();
+      if( 0 > fff.getRemainingFillPower(map, unit.model.movePower, from, to) )
```

Comment:
Do we really need to monitor move power here? This function is just checking if the path goes through something the unit can't go through. Actually, we just call `getRemainingFillPower` without ever updating fill power, so we aren't even monitoring move distance. Is that intentional? If so, it's misleading anyway.
In fact, since `includeOccupiedSpaces` is true, it appears that this wouldn't detect trying to walk through enemies.

Reply:
We're not monitoring move power, really. Yeah, I probably should add a comment.
`getRemainingFillPower()` is being used to determine traversability... which includes being blocked by enemy units in the usual case (but we could totally add a new movetype that goes through enemies and it would work just fine).
`includeOccupiedSpaces` is *only* for the "can I validly end up here" check, not the traversability check (and I am open for suggestions on better nomenclature).

Reply:
If we aren't monitoring move power, why bring in FFF at all? The new formulation for this path-collision check is more verbose and less clear in intent than the old formulation.

Reply:
The reasoning is that FFF is now the 'canonical' way of finding move cost, which creates a new consistent branch point for extensibility.
If I wanted to make a unit type that can move through other units right now, I could. It would work transparently through these functions because they no longer explicitly check for enemy units, but work through the FFF.

If you'd like the FFF function signatures to change in some way to make things more clear, that's fine, but ideally the MoveType should no longer be used directly for practical purposes.
Wally doesn't use it for his analysis, but I'd have to do a lot of meddling to make that work well anyhow.

Reply:
grumble grumble fine.